lms/djangoapps/certificates/signals.py

```diff
@@ -64,3 +65,36 @@ def toggle_self_generated_certs(course_key, course_self_paced):
     """
     course_key = CourseKey.from_string(course_key)
     CertificateGenerationCourseSetting.set_enabled_for_course(course_key, course_self_paced)
+
+
+@receiver(LEARNER_COURSE_PASSING_GRADE, dispatch_uid="new_passing_learner")
```

Comment:
Note: the way this is currently implemented, this will fire a new task every time a passing user submits a problem.  We should update the signal so it fires only when the user's grade changes from passing to non-passing.  
Alternatively, you can check whether the certificate already exists before firing the task.  I don't remember what was originally designed.  But, this logic should be corrected before enabling.